aiida/tools/importexport/common/zip_folder.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Export a zip-file."""
+# pylint: disable=missing-docstring,redefined-builtin
+import io
+import os
+import zipfile
+
+
+class MyWritingZipFile:
+
+    def __init__(self, zip_file, fname):
+        self._zipfile = zip_file
+        self._fname = fname
+        self._buffer = None
+
+    def open(self):
+        if self._buffer is not None:
+            raise IOError('Cannot open again!')
+        self._buffer = io.StringIO()
+
+    def write(self, data):
+        self._buffer.write(data)
+
+    def close(self):
+        self._buffer.seek(0)
+        self._zipfile.writestr(self._fname, self._buffer.read())
+        self._buffer = None
+
+    def __enter__(self):
+        self.open()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.close()
+
+
+class ZipFolder:
+    """
+    To improve: if zipfile is closed, do something
```

Comment:
The current docstring explains what remains to be fixed rather than explaining the purpose/usage of the class

Reply:
Also this class has many functions without docstrings